tap_aggregator/src/main.rs

```diff
@@ -95,3 +124,21 @@ async fn main() -> Result<()> {
     debug!("Goodbye!");
     Ok(())
 }
+
+fn create_eip712_domain(args: &Args) -> Eip712Domain {
+    // Transfrom the args into the types expected by Eip712Domain::new().
+
+    // Transform optional strings into optional Cow<str>.
+    let name = args.domain_name.clone().map(Cow::Owned);
+    let version = args.domain_version.clone().map(Cow::Owned);
+
+    // Transform optional strings into optional U256.
+    let chain_id: Option<U256> = args.domain_chain_id.as_ref().map(|s| s.parse().unwrap());
+    let salt: Option<FixedBytes<32>> = args.domain_salt.as_ref().map(|s| s.parse().unwrap());
```

Comment:
To help the user undestand what went wrong.
```suggestion
    let chain_id: Option<U256> = args.domain_chain_id.as_ref().map(|s| s.parse().expect("Domain chain ID"));
    let salt: Option<FixedBytes<32>> = args.domain_salt.as_ref().map(|s| s.parse().expect("Domain salt"));
```

Reply:
I fixed this by propagating the error instead. Let me know if you would prefer an `expect` there.

Reply:
Even better!